src/control/cmd/daos/filesystem.go

```diff
@@ -603,3 +604,39 @@ func (cmd *fsDfuseEvictCmd) Execute(_ []string) error {
 
 	return nil
 }
+
+type fsScanCmd struct {
+	existingContainerCmd
+
+	DirName string `long:"dir-name" short:"n" description:"subdirectory path to scan. Start from container root if not specified."`
+}
+
+func (cmd *fsScanCmd) Execute(_ []string) error {
```

Comment:
json outout here would be really useful, again that ties in with having the C function pass out a results struct, and presumably using one of the flags to specify no progress logging.